tsdb/querier.go

```diff
@@ -123,6 +127,27 @@ func (q *querier) Select(ms ...*labels.Matcher) (SeriesSet, error) {
 	return NewMergedSeriesSet(ss), nil
 }
 
+func (q *querier) Exemplars(exemplarStorage ExemplarStorage, ms ...*labels.Matcher) ([][]exemplar.Exemplar, error) {
```

Comment:
This is weird, why is the tsdb's queries coupled to looking up exemplars?
This can all be implemented via public interfaces of the querier.

Reply:
Agreed, it can be. This PR is just me modifying the other one to store exemplars without messing with any of the TSDB internal storage types, again as just an example of what the storage could look like.